agents/timestream_aggregator/timestream_aggregator.py

```diff
@@ -217,10 +229,37 @@ def read_frames(self):
             if self.reader is None:
                 return
 
+        flowcontrol_types = [core.G3FrameType.none,
+                             core.G3FrameType.Observation,
+                             core.G3FrameType.Wiring]
+
         self.frames = self.reader.Process(None)
         if self.frames:
-            # Discard flow control frames
-            self.frames = [x for x in self.frames if x.type != core.G3FrameType.none]
+            # Handle flow control frames
+            for f in self.frames:
+                if f.type in flowcontrol_types:
+                    if 'sostream_flowcontrol' not in f:
+                        self.log.warn("Improperly formatted flow control " +
```

Comment:
A standard observation frame wouldn't get discarded since the discard only occurs if 'sostream_flowcontrol' is in the frame. I was just working on this though, since that would mean none type frames that are poorly formatted would make it into the timestream, which we don't want, right? I was going to also add in the previous line that dropped flowcontrol frames:
```
self.frames = [x for x in self.frames if x.type != core.G3FrameType.none]
```

It is a good point that standard Observation and Wiring frames would generate this warning though.

Reply:
I think some type checking will still be required if we want to warn about poorly formatted none type frames, as suggested earlier: https://github.com/simonsobs/socs/pull/46#discussion_r383522174

Reply:
Right, I mis-read the handling of standard Obs+Wiring.

My suggestion in comment #46 was predicated on a question I posed to which I sort of hoped the answer was "no".  My suggestion is: since we now can't use type as the sole means of separating flowcontrol from non-flowcontrol frames (because of CLEANSE), then perhaps we should use the 'sostream_flowcontrol' key as a single criterion.  This would simplify all the logic.  Would it mean that we now consider None-type frames without sostream_flowcontrol to be valid data that should be written into the output stream?  Yes.  Since we don't currently use those, this has no real effect.  But it leaves us the option of using None frames in our real data stream, later.  And more usefully, eliminating the flowcontrol_types list also allows us to add CLEANSE frames of other types, later, without modifying the smurf-recorder.  In case we some day realize we need to incorporate, and thus also to CLEANSE, InstrumentStatus or Calibration or Housekeeping type frames.

Reply:
But... I blocked this merge based on my mis-reading of Obs+Wiring handling.  So I'll "approve" and leave the above as a suggestion / opportunity to say "I knew this would happen!" in 18 months :)

Reply:
Heh, well I took your advice and removed the type checking and check only for the 'sostream_flowcontrol' key.

I fixed it over here, https://github.com/simonsobs/socs/pull/50/commits/9d8b74c536ca5d70be11ab6bf86eb595fd0b5a18, since the tests are present there and that involved moving the location of the code in question (so working on this branch would cause some difficulties.) I'm going to merge that in to this branch now anyway. I kept it separate so that the diff here wouldn't change dramatically while people were reviewing.